File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        vector<pair<int,char>> characters(62,{0,0});
        for(char x: s){
            if(x>='a' && x<='z'){
                characters[x-'a'].second=x;
                characters[x-'a'].first++;
            }else if(x>='A' && x<='Z'){
                characters[x-'A'+26].first++;
                characters[x-'A'+26].second=x;
            }else if(x>='0' && x<='9'){
                characters[x-'0'+52].first++;
                characters[x-'0'+52].second=x;
            }   
        }
        
        sort(characters.begin(), characters.end(), [](pair<int,char> a, pair<int,char> b){
            return a.first>b.first;
        });
        
        string ans="";
        for(auto data: characters){
            if(data.first>0){
                int count=data.first;
                cout<<data.second<<" "<<count<<endl;
                while(count--)ans+=data.second;
            }
        }
        return ans;
    }
};

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution(object):
    def frequencySort(self, s):
        hashmap={}
        for i in s:
            if i in hashmap:
                hashmap[i]+=1
            else:
                hashmap[i]=1
        # return hashmap
        # sort hashmap
        st=sorted(hashmap.keys(),key=lambda x: hashmap[x],reverse=True)
        res=[]
        for i in st:
            res.append(i*hashmap[i])
        return "".join(res)